ping/h3ping.cpp

```diff
@@ -0,0 +1,304 @@
+    printf("  -h, --help             Print this help text\n");
+    printf("  -i, --interval <ms>    Interval between requests in milliseconds (default=1000)\n");
+    printf("  -p, --path <path>      Path to request (default=/)\n");
+    printf("  -t, --timeout <ms>     Timeout for each request in milliseconds (default=5000)\n");
+    printf("  -u, --unsecure         Allow unsecure connections\n");
+    printf("  -v, --verbose          Enable verbose output\n");
+    printf("  -V, --version          Print version information\n");
+}
+
+void ParseArgs(int argc, char **argv) {
+    // Check for version and help first, before requiring server argument
+    for (int i = 1; i < argc; ++i) {
+        if (!strcmp(argv[i], "-?") || !strcmp(argv[i], "-h") || !strcmp(argv[i], "--help")) {
+            PrintUsage(argv[0]);
+            exit(0);
+        } else if (!strcmp(argv[i], "--version") || !strcmp(argv[i], "-V")) {
+            uint32_t Version[4]; MsH3Version(Version);
+            printf("h3ping using msh3 v%u.%u.%u.%u\n", Version[0], Version[1], Version[2], Version[3]);
+            exit(0);
+        }
+    }
+    
+    if (argc < 2) {
+        PrintUsage(argv[0]);
+        exit(0);
+    }
+
+    // Parse the server[:port] argument.
+    Args.Host = argv[1];
+    char *port = strrchr(argv[1], ':');
+    if (port) {
+        *port = 0; port++;
+        Args.Address.SetPort((uint16_t)atoi(port));
+    }
+
+    // Parse options.
+    for (int i = 2; i < argc; ++i) {
+        if (!strcmp(argv[i], "--count") || !strcmp(argv[i], "-c")) {
+            if (++i >= argc) { printf("Missing count value\n"); exit(-1); }
+            Args.Count = (uint32_t)atoi(argv[i]);
+            if (Args.Count == 0) Args.Infinite = true;
+
+        } else if (!strcmp(argv[i], "--interval") || !strcmp(argv[i], "-i")) {
+            if (++i >= argc) { printf("Missing interval value\n"); exit(-1); }
+            Args.Interval = (uint32_t)atoi(argv[i]);
+
+        } else if (!strcmp(argv[i], "--path") || !strcmp(argv[i], "-p")) {
+            if (++i >= argc) { printf("Missing path value\n"); exit(-1); }
+            Args.Path = argv[i];
+
+        } else if (!strcmp(argv[i], "--timeout") || !strcmp(argv[i], "-t")) {
+            if (++i >= argc) { printf("Missing timeout value\n"); exit(-1); }
+            Args.Timeout = (uint32_t)atoi(argv[i]);
+
+        } else if (!strcmp(argv[i], "--unsecure") || !strcmp(argv[i], "-u")) {
+            Args.Flags |= MSH3_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION;
+
+        } else if (!strcmp(argv[i], "--verbose") || !strcmp(argv[i], "-v")) {
+            Args.Verbose = true;
+            
+        } else {
+            printf("Unknown option: %s\n", argv[i]);
+            PrintUsage(argv[0]);
+            exit(-1);
+        }
+    }
+}
+
+bool SendPingRequest(MsH3Connection& Connection) {
+    MSH3_HEADER Headers[] = {
+        { ":method", 7, "HEAD", 4 },  // Use HEAD for minimal overhead
```

Comment:
Add another cmd line parameter to make this a GET if desired. Update all documentation as well.